### PR TITLE
feat(ST-49): add get classrooms for students at the existing api

### PR DESCRIPTION
**Pull request by faysal-alam-sazim**
## Description of Change

- Added student's classroom fetching logic inside getClassrooms function

## Associated Ticket Link(s)

- https://trello.com/c/w6Kj5pxf

## Related Pull Request(s)

- N/A

### CODE REVIEW

File: src/classrooms/classrooms.service.ts [hunk 1]
```diff
@@ -25,10 +25,18 @@ export class ClassroomsService {
...
     }
-    // return student's enrolled classrooms
-    return [];
+
+    const classroomStudents = await this.classroomsStudentsRepository.find(
+      { studentId: user.student },
+      { populate: ["classroomId"] },
+    );
+
+    const classrooms = classroomStudents.map((classroomStudent) => classroomStudent.classroomId);
```
Comment (ishmam-iqbal-sazim) [inline src/classrooms/classrooms.service.ts:37]: Won't you just be getting an array of classroom ids here?

Comment (faysal-alam-sazim) [reply src/classrooms/classrooms.service.ts:37]: ClassroomId is mapped to Classroom entity. So it is returning array of classrooms.

Comment (ishmam-iqbal-sazim) [reply src/classrooms/classrooms.service.ts:37]: Please change the code such that it's more readable. This implementation can be confusing

File: src/classrooms/classrooms.service.ts [hunk 2]
```diff
@@ -25,10 +25,19 @@ export class ClassroomsService {
 
     if (role === EUserRole.TEACHER) {
       const classrooms = await this.classroomsRepository.find({ teacher: user.teacher });
+
       return classrooms;
```
Comment (ishmam-iqbal-sazim) [inline src/classrooms/classrooms.service.ts:29]: This service should always return classrooms, no need to return in this `if` block

File: src/classrooms/classrooms.service.ts [hunk 3]
```diff
@@ -25,10 +25,19 @@ export class ClassroomsService {
...
-    // return student's enrolled classrooms
-    return [];
+
+    const classroomStudents = await this.classroomsStudentsRepository.find({
+      studentId: user.student,
+    });
+
+    const classroomIds = classroomStudents.map((classroom) => classroom.classroomId.id);
+
+    const classrooms = await this.classroomsRepository.find({ id: { $in: classroomIds } });
```
Comment (ishmam-iqbal-sazim) [inline src/classrooms/classrooms.service.ts:38]: Should wrap this around a role check where if the user is a student, it assigns classrooms the result of the find value

Comment (ishmam-iqbal-sazim) [inline src/classrooms/classrooms.service.ts:38]: Can initialize a classrooms var at the top of the method and assign the value here

**Review by ishmam-iqbal-sazim — APPROVED** [review]
Left some comments

**Pull request merged**